pkg/services/ngalert/migration/alert_rule.go

```diff
@@ -287,17 +301,19 @@ func truncate(daName string, length int) string {
 	return daName
 }
 
-func extractChannelIDs(d *migrationStore.DashAlert) (channelUids []migrationStore.UidOrID) {
+// extractChannelUIDs extracts the notification channel UIDs from the given legacy dashboard alert parsed settings.
+func (om *OrgMigration) extractChannelUIDs(ctx context.Context, l log.Logger, orgID int64, parsedSettings dashAlertSettings) (channelUids []string) {
 	// Extracting channel UID/ID.
-	for _, ui := range d.ParsedSettings.Notifications {
-		if ui.UID != "" {
-			channelUids = append(channelUids, ui.UID)
-			continue
-		}
-		// In certain circumstances, id is used instead of uid.
-		// We add this if there was no uid.
+	for _, ui := range parsedSettings.Notifications {
+		// Either id or uid can be defined in the dashboard alert notification settings. See alerting.NewRuleFromDBAlert.
 		if ui.ID > 0 {
-			channelUids = append(channelUids, ui.ID)
+			uid, err := om.migrationStore.GetAlertNotificationUidWithId(ctx, orgID, ui.ID)
+			if err != nil {
+				l.Error("Failed to get alert notification UID", "notificationId", ui.ID, "err", err)
+			}
```

Comment:
I think this answers my question above: we are going to drop a notification without UID. I wonder what is the reason for such decision?

Reply:
If neither `id` not `uid` is present on the entry, or if the `id` is invalid, then that legacy alert would be failing to send to the channel in legacy alerting anyways.